node/pkg/allocateip/allocateip.go

```diff
@@ -565,7 +564,7 @@ func removeHostTunnelAddr(ctx context.Context, c client.Interface, nodename stri
 					// No allocation exists, we don't have anything to do.
 				} else if len(attr) == 0 && handle == nil {
 					// The IP is ours. Release it by passing the exact IP.
-					if _, err := c.IPAM().ReleaseIPs(ctx, []net.IP{*ipAddr}); err != nil {
+					if _, err := c.IPAM().ReleaseIPs(ctx, ipam.ReleaseOptions{Address: ipAddr.String()}); err != nil {
```

Comment:
Is there a reason we can't use the new approach here with IP and handle both passed in?  I guess the allocation could predate handles?

Reply:
Hm, let me look at this. 

I think there are some complex edge cases here. There are several different scenarios that this code tries to deal with and I can't recall exactly how all of them work.

Reply:
So there are a few things here:

- The old allocation could pre-date the use of handles. We have some special code to handle upgrading those to use handles, but I think it's still possible we hit this branch without a handle (which trickles through and is handled below).
- We do actually want to release all IPs with the tunnel address handle (if there are more than one) in this codepath, because if there are more than one it means somehow we leaked an address at some point.

That said, I think we _do _want to catch the case that the GC was encountering. Namely, that the allocation still exists but the handle object doesn't. 

I've added an additional branch to this section to catch that.